src/UID2.Client/UID2Encryption.cs

```diff
@@ -25,19 +25,20 @@ internal static DecryptionResponse Decrypt(string token, KeyContainer keys, Date
             string headerStr = token.Substring(0, 4);
             Boolean isBase64UrlEncoding = headerStr.IndexOfAny(BASE64_URL_SPECIAL_CHARS) != -1;
             byte[] data = isBase64UrlEncoding ? UID2Base64UrlCoder.Decode(headerStr) : Convert.FromBase64String(headerStr);
+            IdentityType? identityType = GetIdentityType((byte) ((data[0] & 12) >> 2));
```

Comment:
Maybe use a binary literal? 0b1100?

Reply:
Also why 2 bits if it's just 0 or 1? Is the mask just 0b100?

Reply:
I feel like doing the check here doesn't feel right. For V2 it doesn't mean anything. What if for V5+ this bit is moved? Maybe this logic should just be moved to DecryptV3 and casted to IdentityType straight away, no need for the GetIdentityType function if possible

Reply:
I think specifically with "A"/"B"/"E"/"F" in hexadecimal representation, after we convert that to binary and perform the AND operation against 1100, you will always get 0X00. So you just shift right twice to get 000X.